File: app/schemas/job.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.models.job_status import JobStatus


class JobResponse(BaseModel):
    id: str
    filename: str

    status: JobStatus
    progress: int

    language: Optional[str] = None
    duration: Optional[float] = None

    created_at: datetime
    completed_at: Optional[datetime] = None

    pdf_path: Optional[str] = None
    txt_path: Optional[str] = None
    json_path: Optional[str] = None

    error: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/job.ts
export type JobStatus =
  | 'UPLOADED'
  | 'EXTRACTING_AUDIO'
  | 'DETECTING_SPEECH'
  | 'TRANSCRIBING'
  | 'MERGING'
  | 'GENERATING_PDF'
  | 'COMPLETED'
  | 'FAILED';

export interface Job {
  id: string;
  filename: string;
  status: JobStatus;
  progress: number;
  language: string | null;
  duration: number | null;
  created_at: string;
  completed_at: string | null;
  pdf_path: string | null;
  txt_path: string | null;
  json_path: string | null;
  error: string | null;
}

export interface UploadResponse {
  message: string;
  filename: string;
  path: string;
}

export interface TranscribeResponse {
  message: string;
  job_id: string;
  status: JobStatus;
  progress: number;
}

export interface MessageResponse {
  success: boolean;
  message: string;
}
